code-challenges.js

```diff
@@ -39,3 +146,20 @@ var numbersToAdd2 = [0, 7, -8, 12]
 // Expected output: [0, 7, -1, 11]
 var numbersToAdd3 = []
 // Expected output: []
+
+//create a function that takes in an array and adds all the previous numbers to each following numbers
+const addNumbersArray = (array) => {
+  // map over the array & return a new array including the mapped values
+return array.map((value, index) => {
+  // if the value is AT index of 0
+  //I tried so many ways to write this but kept getting NaN as answers.... I want it to check if the value is at the 0 index in the array, and if it is, I want the returned value to remain the same
+    if(index === 0){
+      //for the value at index 0, return the original value
+    return value
+    }else{
+  //for all values at later positions in the array, return the original value + the value at the previous index.
+  return value + value[index - 1]
+    }
+  })
+}
+console.log(addNumbersArray(numbersToAdd1))
```

Comment:
You have all the logic. The issue looks to be relying on the value inside map to add the subsequent values. Every time map iterates, value gets reset. So establishing an anchor outside of the map will get you what you need.